cli/azd/cmd/provision.go

```diff
@@ -129,34 +130,32 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 		defaultTitleNote = "This is a preview. No changes will be applied to your Azure resources."
 	}
 
+	p.console.MessageUxItem(ctx, &ux.MessageTitle{
+		Title:     defaultTitle,
+		TitleNote: defaultTitleNote},
+	)
+
 	// Get Subscription to Display in Command Title Note
 	subscriptions, subErr := p.subManager.GetSubscriptions(ctx)
 	if subErr == nil {
 		// Find subscription name
 		for _, sub := range subscriptions {
 			if sub.Id == p.env.GetSubscriptionId() {
-				messageFormat := "Provisioning Azure resources in subscription (%s) %s and location (%s) can take some time"
-				if previewMode {
-					messageFormat = "This is a preview. No changes will be applied to your Azure resources in subscription (%s) %s " +
-						"and location (%s)."
+				location, err := p.subManager.GetLocation(ctx, p.env.GetSubscriptionId(), p.env.GetLocation())
+				if err != nil {
+					log.Printf("failed getting location: %v", err)
 				}
-				// Formate the note
-				defaultTitleNote = fmt.Sprintf(
-					messageFormat,
-					sub.Name,
-					sub.Id,
-					p.env.GetLocation(),
+				p.console.MessageUxItem(ctx, &ux.EnvironmentDetails{
+					Subscription: fmt.Sprintf("%s (%s)", sub.Name, sub.Id),
+					Location:     location.DisplayName},
 				)
 				break
 			}
 		}
```

Comment:
It feels like we should get ride of this loop and add a function on `SubscriptionManager` to do a lookup on subscription id.

Reply:
done